test/k8s/datapath_configuration.go

```diff
@@ -453,168 +437,9 @@ var _ = Describe("K8sDatapathConfig", func() {
-				res2, cancel2, err := kubectl.ExecInHostNetNSInBackground(context.TODO(), k8s2NodeName, cmd)
-				ExpectWithOffset(2, err).Should(BeNil(), "Cannot exec tcpdump in bg")
-
-				// HTTP connectivity test (pod2pod)
-				kubectl.ExecPodCmd(randomNamespace, srcPod,
-					helpers.CurlFail("http://%s/", net.JoinHostPort(dstIP, "80"))).ExpectSuccess("Failed to curl dst pod")
-
-				// Check that no unencrypted pod2pod traffic was captured on the direct routing device
-				cancel1()
-				cancel2()
-				ExpectWithOffset(2, res1.CombineOutput().String()).Should(Not(ContainSubstring("1 packet captured")))
-				ExpectWithOffset(2, res2.CombineOutput().String()).Should(Not(ContainSubstring("1 packet captured")))
-			}
-
-			checkNoLeak(srcPod, srcPodIP.String(), dstPodIP.String())
-			checkNoLeak(srcPod, srcPodIPv6, dstPodIPv6)
-
-			// Check that the src pod can reach the remote host
-			kubectl.ExecPodCmd(randomNamespace, srcPod, helpers.Ping(k8s2IP)).
-				ExpectSuccess("Failed to ping k8s2 host from src pod")
-
-			// Check that the remote host can reach the dst pod
-			kubectl.ExecInHostNetNS(context.TODO(), k8s1NodeName,
-				helpers.CurlFail("http://%s:80/", dstPodIP)).ExpectSuccess("Failed to curl dst pod from k8s1")
-		}
-
-		It("Pod2pod is encrypted in direct-routing mode", func() {
-			deploymentManager.DeployCilium(map[string]string{
-				"tunnel":               "disabled",
-				"autoDirectNodeRoutes": "true",
-				"encryption.enabled":   "true",
-				"encryption.type":      "wireguard",
-				"l7Proxy":              "false",
-			}, DeployCiliumOptionsAndDNS)
-
-			privateIface, err := kubectl.GetPrivateIface(helpers.K8s1)
-			Expect(err).Should(BeNil(), "Cannot determine private iface")
-			testWireguard(privateIface)
-		})
-
-		It("Pod2pod is encrypted in tunneling mode", func() {
-			deploymentManager.DeployCilium(map[string]string{
-				"tunnel":             "vxlan",
-				"encryption.enabled": "true",
-				"encryption.type":    "wireguard",
-				"l7Proxy":            "false",
-			}, DeployCiliumOptionsAndDNS)
-
-			testWireguard("cilium_vxlan")
-		})
-
-		It("Pod2pod is encrypted in tunneling mode with per-endpoint routes", func() {
-			deploymentManager.DeployCilium(map[string]string{
-				"tunnel":                 "vxlan",
-				"endpointRoutes.enabled": "true",
-				"encryption.enabled":     "true",
-				"encryption.type":        "wireguard",
-				"l7Proxy":                "false",
-			}, DeployCiliumOptionsAndDNS)
-
-			testWireguard("cilium_vxlan")
-		})
-
-	})
-
 	SkipContextIf(func() bool {
 		return helpers.RunsOnGKE() || helpers.RunsWithoutKubeProxy() || helpers.RunsOnAKS()
 	}, "Transparent encryption DirectRouting", func() {
-		var privateIface string
-		BeforeAll(func() {
-			Eventually(func() (string, error) {
-				iface, err := kubectl.GetPrivateIface(helpers.K8s1)
-				privateIface = iface
-				return iface, err
-			}, helpers.MidCommandTimeout, time.Second).ShouldNot(BeEmpty(),
-				"Unable to determine private iface")
-		})
```

Comment:
I think this was added to avoid a flake in the device detection, so probably best to keep it for the second `It` block.